src/ucp/tag/tag_send.c

```diff
@@ -292,7 +292,7 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_tag_send_nbx,
                                         datatype, contig_length, param);
     } else {
         ucp_tag_send_req_init(req, ep, buffer, datatype, memory_type, count,
-                              tag, 0);
+                              tag, req->flags & UCP_REQUEST_FLAG_EXTERNAL);
```

Comment:
why need reinit?

Reply:
this is not re-init, req_init fills all values in request

Reply:
req->flags is already initialized, why need to set it again? can just do `|=` in req_init

Reply:
there added more flags